src/writer/workflows_blocks/githubintegration.py

```diff
@@ -0,0 +1,436 @@
+                return {
+                    "content": content,
+                    "sha": contents.sha,
+                    "size": contents.size
+                }
+            elif operation == "create":
+                result = repo.create_file(
+                    path=file_path,
+                    message=commit_message,
+                    content=content
+                )
+                return {
+                    "commit": {
+                        "sha": result["commit"].sha,
+                        "message": result["commit"].message
+                    },
+                    "content": {
+                        "path": result["content"].path,
+                        "sha": result["content"].sha
+                    }
+                }
+            elif operation == "update":
+                contents = repo.get_contents(file_path)
+                result = repo.update_file(
+                    path=file_path,
+                    message=commit_message,
+                    content=content,
+                    sha=contents.sha
+                )
+                return {
+                    "commit": {
+                        "sha": result["commit"].sha,
+                        "message": result["commit"].message
+                    },
+                    "content": {
+                        "path": result["content"].path,
+                        "sha": result["content"].sha
+                    }
+                }
+            elif operation == "delete":
+                contents = repo.get_contents(file_path)
+                result = repo.delete_file(
+                    path=file_path,
+                    message=commit_message,
+                    sha=contents.sha
+                )
+                return {
+                    "commit": {
+                        "sha": result["commit"].sha,
+                        "message": result["commit"].message
+                    }
+                }
+        except GithubException as e:
+            self.outcome = "error"
+            raise RuntimeError(f"File operation error: {str(e)}")
+
+    def run(self):
+        try:
+            # Get authentication token
+            token = self._get_field("token")
+            self._initialize_client(token)
+
+            # Get operation details
+            operation = self._get_field("operation")
+            repo_name = self._get_field("repo_name", True)
+            owner = self._get_field("owner", True)
+            branch = self._get_field("branch", True)
+            file_path = self._get_field("file_path", True)
+            content = self._get_field("content", True)
+            title = self._get_field("title", True)
+            body = self._get_field("body", True)
+            labels = json.loads(self._get_field("labels", True, "[]"))
+            assignees = json.loads(self._get_field("assignees", True, "[]"))
+            commit_message = self._get_field("commit_message", True)
```

Comment:
Some fields are not JSON, and some are parsed twice
```suggestion
            operation = self._get_field("operation")
            repo_name = self._get_field("repo_name")
            owner = self._get_field("owner")
            branch = self._get_field("branch")
            file_path = self._get_field("file_path")
            content = self._get_field("content")
            title = self._get_field("title")
            body = self._get_field("body")
            labels = self._get_field("labels", True, "[]")
            assignees = self._get_field("assignees", True, "[]")
            commit_message = self._get_field("commit_message")
```